### PR TITLE
Implement delete downloaded feature

### LINKED ISSUE #5: Delete downloaded questions
Allow users to delete the downloaded questions (to save some bytes maybe ?).

### PR DESCRIPTION
**Pull request by Aqua-16**
## Description

Made a button for deleting from downloaded list. Implement function to delete from downloaded JSON list. Also update the page of each downloaded question after deletion.

## Related Issue

Fixes #5 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by shank03** [comment]
@Aqua-16 Based on inactivity of 6 hours, I think you forgot to make PR request on ContriHUB ?

**Comment by Aqua-16** [comment]
Damn, yes I did. Let me fix that.

**Comment by shank03** [comment]
@Aqua-16 It's working great. It's nice to see how you have utilized the existing logic from downloading question. However there are few things:
1. Why is the `PAGE_LIMIT` changed from 25 to 22 ?
2. Why `deleteDownloadedQuestions` require `NetworkClient.withNetwork` ?
3. Why is `DownloadManager` responsible for `deleteQuestion` when it should be `AppData` ?
4. I can see that `downloadQuestions` and `deleteDownloadedQuestions` have very similar codes. Think you can break it into better code ? (you'll get bonus for this)

**Comment by Aqua-16** [comment]
I changed `PAGE_LIMIT` to 22 as it fits questions on the page without introducing the scroll bar.
I tried without the network thing, but it didn't work as expected. Maybe I made some error back then, so I'll try without it once more.
I kept deleteQuestion in DownloadManager because then all details related to Downloaded Questions are handled by one file alone, be that adding new downloaded questions, or deleting previous ones.

And I'll try to merge the functions, definitely looks possible.

**Comment by shank03** [comment]
@Aqua-16 
> I changed `PAGE_LIMIT` to 22 as it fits questions on the page without introducing the scroll bar. 

Well, don't you think this number would vary based on screen density and window size ? hence, it's better not to 
change it continously.

> I kept deleteQuestion in DownloadManager because then all details related to Downloaded Questions are handled by one file alone, be that adding new downloaded questions, or deleting previous ones.

Then how about we rename it to `QuestionManager` ? or `ProblemManager` ?

Also, can you optimize `DownloadManager.deleteQuestion` function ? I can see the iterations in `for` could serve better.

**Comment by Aqua-16** [comment]
I'll be renaming it to QuestionManager.
And you're right. Since it depends on screen size, I'll just default to 25 as its a prettier number.

I'm also making some changes not really mentioned in the issue itself like download text being stuck "Download n Questions" since that was bugging me.

**Comment by Aqua-16** [comment]
Reduced time from N*M to N+M with extra space in deleteQuestion.

File: src/main/java/com/shank/offcoder/cf/QuestionManager.java [hunk 1]
```diff
@@ -74,21 +75,17 @@ public static void deleteQuestion(Controller controller, final List<ProblemParse
...
-                    }
-                }
-                if (flag) {
-                    questionIndex.add(idx - ((int) counter - 1));
+            for (int i = 0; i < arr.length(); i++) {
+                JSONObject item = arr.getJSONObject(i);
+                if (pCode.contains(item.get(AppData.P_CODE_KEY).toString())) {
+                    ++counter;
+                    controller.downloadProgress.setProgress(counter / list.size());
+                    questionIndex.add(i);
```
Comment (shvdx) [inline src/main/java/com/shank/offcoder/cf/QuestionManager.java]: @Aqua-16 how about ~`questionIndex.add(i)`~ `arr.remove(i)` here itself ?

**Review by shank03 — CHANGES_REQUESTED** [review]

File: src/main/java/com/shank/offcoder/cf/QuestionManager.java (hunk 1 above)
Comment (Aqua-16) [reply src/main/java/com/shank/offcoder/cf/QuestionManager.java]: Won't that cause an error? I haven't tried that yet, but doing something like that usually gives me an error. Removing element at index, while iterating that is

Comment (Aqua-16) [reply src/main/java/com/shank/offcoder/cf/QuestionManager.java]: Alright, tested it and there were no errors. I think that error was raised while working with HashSets and I just generalized it to all structures.

**Review by shank03 — APPROVED** [review]

**Pull request merged**